File: finstack-py/tests/golden/runners/calibration_vol_smile.py
"""Domain runner for equity and FX volatility smile calibration goldens."""

from __future__ import annotations

from tests.golden.runners._placeholders import reject_flattened_outputs
from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Require executable vol smile calibration inputs before running."""
    return reject_flattened_outputs("vol smile calibration runner", fixture)

File: finstack/valuations/tests/golden/runners/calibration_vol_smile.rs
//! Domain runner for equity and FX volatility smile calibration golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Volatility smile calibration golden runner.
pub struct CalibrationVolSmileRunner;

impl DomainRunner for CalibrationVolSmileRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::reject_flattened_outputs("vol smile calibration runner", fixture)
    }
}
